modules/renderer/features.js

```diff
@@ -412,6 +412,18 @@ export function rendererFeatures(context) {
     };
 
 
+    features.isHiddenPreset = function(preset, geometry) {
+        if (!_hidden.length) return false;
+        if (!preset.tags) return false;
+        for (var i = 0; i < _hidden.length; i++) {
+            if (_features[_hidden[i]].filter(preset.tags, geometry)) {
```

Comment:
Probably want to use `addTags` (the ones that will get added) instead of `tags` (the ones that will get matched).  They are usually the same, but not necessarily so.

Also, this might not work for situations where applying a preset will set a tag because of a field.  So if you are hiding buildings, and the preset has `building_area` field (for example "Office"), and you draw an area, that feature will get a `building=yes` tag assigned.

Reply:
Good point. I just used `setTags` with `{}` to get `addTags` and the field defaults.

Reply:
nice 👍